api/v1alpha1/edgedeployment_webhook.go

```diff
@@ -93,6 +96,13 @@ func (r *EdgeDeployment) validate() error {
 				}
 			}
 		}
+
+		if _, ok := containersNames[container.Name]; ok {
+			return fmt.Errorf("Name collisions for containers within the same pod spec are not supported.\n" +
+				"Container name: '%s' has been reused.", container.Name)
+		} else {
+			containersNames[container.Name] = name
```

Comment:
and here with `struct{}{}` instead of `name`

Reply:
done